src/Sylius/Bundle/ResourceBundle/Controller/ResourcesCollectionProvider.php

```diff
@@ -48,12 +48,19 @@ public function __construct(ResourcesResolverInterface $resourcesResolver, Pager
     public function get(RequestConfiguration $requestConfiguration, RepositoryInterface $repository)
     {
         $resources = $this->resourcesResolver->getResources($requestConfiguration, $repository);
+        $paginations = [];
 
-        $paginator = $resources instanceof ResourceGridView ? $resources->getData() : $resources;
+        if ($resources instanceof ResourceGridView) {
+            $paginator = $resources->getData();
+            $paginations = $resources->getDefinition()->getPagination();
+        } else {
+            $paginator = $resources;
+        }
 
         if ($paginator instanceof Pagerfanta) {
             $request = $requestConfiguration->getRequest();
-            $paginator->setMaxPerPage($requestConfiguration->getPaginationMaxPerPage());
+
+            $paginator->setMaxPerPage($request->query->get('paginate', reset($paginations) ?: $requestConfiguration->getPaginationMaxPerPage()));
```

Comment:
Always was confused why this was different for paginated resources. Would like to see consistency in using `limit` for paginated resources too.